fw/http_parser.c

```diff
@@ -2238,13 +2321,22 @@ __req_parse_authorization(TfwHttpReq *req, unsigned char *data, size_t len)
 static int
 __req_parse_cache_control(TfwHttpReq *req, unsigned char *data, size_t len)
 {
-	int r = CSTR_NEQ;
+	/* Very similar to __resp_parse_cache_control */
+	int r = TFW_BLOCK;
 	__FSM_DECLARE_VARS(req);
 
 	__FSM_START(parser->_i_st);
 
+	__FSM_STATE(Req_I_CC_start) {
+		parser->cache_control.dir_flag = 0;
+		parser->cache_control.filled = false;
+		/* A flag can be precluded either by semicolumn
+		 * (implied in this branch) or by comma. */
+		parser->cache_control.dir_allowed = true;
+		/* fall through */
+	}
 	__FSM_STATE(Req_I_CC) {
-		WARN_ON_ONCE(parser->_acc);
+		WARN_ON_ONCE(parser->cache_control.dir_flag);
```

Comment:
This is just not addressed comment https://github.com/tempesta-tech/tempesta/pull/1539/files#r775029875 . Please review the review notes once more. In short: you shouldn't fit new logic into existing states, most likely you need several new states.Introducing new states hurts, but not so much as introducing a lot of complexity with many if's and extra variables.